File: go/isogram/isogram.go
package isogram

import "unicode"

// IsIsogram shows whether word is an isogram
func IsIsogram(word string) bool {
	seen := make(map[rune]int)
	for _, ch := range word {
		if ch == '-' || ch == ' ' {
			continue
		}
		newCh := unicode.ToLower(ch)
		if _, ok := seen[newCh]; ok {
			seen[newCh] += 1
		} else {
			seen[newCh] = 1
		}
	}
	for _, count := range seen {
		if count > 1 {
			return false
		}
	}
	return true
}

File: javascript/isogram/isogram.js
export const isIsogram = (word) => {
  let charMap = Array.from({ length: 26 }).fill(0);
  let normWord = word.toLowerCase().replace(/[\s-]+/, '');
  for (const ch of normWord) {
    charMap[ch.charCodeAt(0) - 'a'.charCodeAt(0)]++;
  }
  return charMap.every(count => count <= 1);
};
